firefox-ios/Client/Frontend/Browser/Tabs/Views/RemoteTabsTableViewController.swift

```diff
@@ -279,6 +295,67 @@ class RemoteTabsTableViewController: UITableViewController,
         remoteTabsPanel?.remoteTabsClientAndTabsDataSourceDidSelectURL(tab.URL, visitType: VisitType.typed)
     }
 
+    override func tableView(_ tableView: UITableView,
+                            commit editingStyle: UITableViewCell.EditingStyle,
+                            forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+            let clientAndTabs = state.clientAndTabs[indexPath.section]
+            guard let fxaDeviceId = clientAndTabs.client.fxaDeviceId else {
+                // This is a state we probably shouldn't be able to get into
+                // so logging the error
+                logger.log("Error retrieving remote device ID to close tab",
+                           level: .fatal,
+                           category: .tabs)
+                return
+            }
+            let tab = clientAndTabs.tabs[indexPath.item]
+
+            // Setting the two private variables below so that the toast button action function has access to them
+            // since that function cannot have any parameters.
+            self.closeTabRemoteDeviceId = fxaDeviceId
+            self.closeTab = tab
+
+            // Creating a modal with an undo button that will allow the user to undo closing the last remote tab
+            // they attempted to close
+            ActionToast(
+                text: SingleTabTitle,
+                bottomContainer: view,
+                theme: themeManager.getCurrentTheme(for: windowUUID),
+                buttonTitle: UndoString,
+                buttonAction: self.undo
+            ).show()
```

Comment:
This PR currently doesn't build because of the strings. I think they should be this instead, right?

```suggestion
            ActionToast(
                text: .TabsTray.CloseTabsToast.SingleTabTitle,
                bottomContainer: view,
                theme: themeManager.getCurrentTheme(for: windowUUID),
                buttonTitle: .TabsTray.CloseTabsToast.Action,
                buttonAction: self.undo
            ).show()
```